Mathlib/Analysis/NormedSpace/OperatorNorm/NormedSpace.lean

```diff
@@ -278,7 +278,17 @@ theorem norm_smulRightL (c : E →L[𝕜] 𝕜) [Nontrivial Fₗ] : ‖smulRight
   ContinuousLinearMap.homothety_norm _ c.norm_smulRight_apply
 #align continuous_linear_map.norm_smul_rightL ContinuousLinearMap.norm_smulRightL
 
-variable (𝕜) (𝕜' : Type*)
+/-- An auxiliary instance to be able to just state the fact that the norm of `smulRightL` makes
+sense. This shouldn't be needed. TODO: fix typeclass inference. -/
+def seminormedAddCommGroup_aux_for_smulRightL :
+    SeminormedAddCommGroup ((E →L[𝕜] 𝕜) →L[𝕜] Fₗ →L[𝕜] E →L[𝕜] Fₗ) :=
```

Comment:
This isn't a max recursion depth issue for synthesis? (Not at a computer)

Reply:
My guess is it's like https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/nested.20linear.20map.20norm.20failure/near/423226292 (`synthPending` issue)

Reply:
Sorry. Yes that was what I was thinking of. Can you just bump it locally?

Reply:
I don't know how to do that (there are no examples throughout mathlib, if I can tell correctly).

Reply:
If you mean `set_option synthInstance.maxHeartbeats 0 in`, it doesn't work here.

Reply:
There is a TODO in the core code which limits the number of pending instance problems when solving for another. It is set to a very small number but bumping it in core broke quite a bit. 

I don't think we are getting any kind of help on this from core in the short term so we should just figure out here how we want to handle it. 

The same problem pops up in `Analysis.NormedSpace.OperatorNorm` where some `letI`'s helped things along. 

Is there anyway to contain the new local instances to the declarations they are needed for?

Reply:
Instead of having a local instance, one can do as follows:
```lean
lemma norm_smulRightL_le :
    letI := seminormedAddCommGroup_aux_for_smulRightL 𝕜 E Fₗ
    ‖smulRightL 𝕜 E Fₗ‖ ≤ 1 :=
```
It's probably a better solution, let me switch to it.